src/D2L.CodeStyle.Analysis/MutabilityInspector.cs

```diff
@@ -89,6 +89,15 @@ ITypeSymbol type
 				return false;
 			}
 
+
+			// make sure we're dealing with the concrete implementation
+			if( type.TypeKind == TypeKind.Interface ) {
+				return true;
+			}
+			if( type.TypeKind == TypeKind.Class && ( !type.IsSealed || type.IsAbstract ) ) {

```

Comment:
I think IsAbstract is irrelevant here.  Removing it, all the tests still pass.  I think we really just care that it isn't sealed